test/HypLSP8.t.sol

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.13;
+
+import { Test } from "forge-std/src/Test.sol";
+
+import { TypeCasts } from "@hyperlane-xyz/core/contracts/libs/TypeCasts.sol";
+import { TestMailbox } from "@hyperlane-xyz/core/contracts/test/TestMailbox.sol";
+import { TestPostDispatchHook } from "@hyperlane-xyz/core/contracts/test/TestPostDispatchHook.sol";
+import { TestInterchainGasPaymaster } from "@hyperlane-xyz/core/contracts/test/TestInterchainGasPaymaster.sol";
+import { GasRouter } from "@hyperlane-xyz/core/contracts/client/GasRouter.sol";
+import { TokenRouter } from "@hyperlane-xyz/core/contracts/token/libs/TokenRouter.sol";
+
+// Mock + contracts to test
+import { HypLSP8 } from "../src/HypLSP8.sol";
+import { HypLSP8Collateral } from "../src/HypLSP8Collateral.sol";
+import { LSP8Mock } from "./LSP8Mock.sol";
+
+abstract contract HypTokenTest is Test {
+    using TypeCasts for address;
+
+    uint256 internal constant INITIAL_SUPPLY = 10;
+    string internal constant NAME = "Hyperlane NFTs";
+    string internal constant SYMBOL = "HNFT";
+
+    address internal ALICE = makeAddr("alice");
+    address internal BOB = makeAddr("bob");
+    address internal OWNER = makeAddr("owner");
+    uint32 internal constant ORIGIN = 11;
+    uint32 internal constant DESTINATION = 22;
+    bytes32 internal constant TOKEN_ID = bytes32(uint256(1));
+    string internal constant URI = "http://example.com/token/";
+
+    LSP8Mock internal localPrimaryToken;
+    LSP8Mock internal remotePrimaryToken;
+    TestMailbox internal localMailbox;
+    TestMailbox internal remoteMailbox;
+    TokenRouter internal localToken;
+    HypLSP8 internal remoteToken;
+    TestPostDispatchHook internal noopHook;
+
+    function setUp() public virtual {
+        localMailbox = new TestMailbox(ORIGIN);
+        remoteMailbox = new TestMailbox(DESTINATION);
+
+        localPrimaryToken = new LSP8Mock(NAME, SYMBOL, OWNER);
+
+        noopHook = new TestPostDispatchHook();
+        localMailbox.setDefaultHook(address(noopHook));
+        localMailbox.setRequiredHook(address(noopHook));
+
+        remoteToken = new HypLSP8(address(remoteMailbox));
+
+        vm.prank(OWNER);
+        remoteToken.initialize(0, NAME, SYMBOL);
+
+        vm.deal(ALICE, 1 ether);
+    }
+
+    function _deployRemoteToken(bool isCollateral) internal {
+        if (isCollateral) {
+            remoteToken = new HypLSP8(address(remoteMailbox));
+            vm.prank(OWNER);
+            remoteToken.initialize(0, NAME, SYMBOL);
+        } else {
+            remoteToken = new HypLSP8(address(remoteMailbox));
+            vm.prank(OWNER);
+            remoteToken.initialize(0, NAME, SYMBOL);
```

Comment:
Suggest putting a local variable above to explain what the `0` is (can find a better variable name.

```solidity
uint256 numberOfTokenIdsToMint = 0;
```

or use named parameters in function call as an alternative:

```suggestion
            remoteToken.initialize({ _mintAmount: 0, _name: NAME, _symbol: SYMBOL });
```